include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1277,7 +1279,8 @@ namespace VectorTools
 
       typedef MatrixFreeOperators::MassOperator<dim, fe_degree, n_q_points_1d, 1, Number> MatrixType;
       MatrixType mass_matrix;
-      mass_matrix.initialize(matrix_free);
+      mass_matrix.initialize(std_cxx11::make_shared<
+                             const MatrixFree<dim,typename VectorType::value_type> > (matrix_free));
```

Comment:
This creates a copy of the `matrix_free` object, right? That is a rather expensive operation. I would prefer to wrap `MatrixFree` into a pointer where the destructor of `shared_ptr` does not release the memory, i.e., by providing a dummy deleter to the constructor of `shared_ptr` that does nothing, see e.g. here: http://stackoverflow.com/questions/20131877/how-do-you-make-stdshared-ptr-not-call-delete

The alternative would be to also expose the `shared_ptr` also in the `project` method, but @davydden is right that it does not need to be that right now.